provisioning/baremetal_pod.go

```diff
@@ -297,9 +308,20 @@ func ipOptionForMachineOsDownloader(info *ProvisioningInfo) string {
 }
 
 func createInitContainerMachineOsDownloader(info *ProvisioningInfo, setIpOptions bool) corev1.Container {
+	imageURL := getMetal3DeploymentConfig(machineImageUrl, &info.ProvConfig.Spec)
 	env := []corev1.EnvVar{
-		buildEnvVar(machineImageUrl, &info.ProvConfig.Spec),
+		{
+			Name:  machineImageUrl,
+			Value: *imageURL,
+		},
```

Comment:
It's not clear to me that this is different from what `buildEnvVar()` does?

Reply:
Just trying to reuse the imageURL variable. I 'll move it back to buildEnvVar. Its more readable.